src/Network/Network/VirtualRouter/GetAzureRMVirtualRouterCommand.cs

```diff
@@ -0,0 +1,125 @@
+    public partial class GetAzureRmVirtualRouter : NetworkBaseCmdlet
+    {
+        [Parameter(
+            ParameterSetName = VirtualRouterParameterSetNames.ByVirtualRouterName,
+            Mandatory = true,
+            HelpMessage = "The resource group name of the virtual router.",
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName")]
+        [Parameter(
+            ParameterSetName = VirtualRouterParameterSetNames.ByVirtualRouterName,
+            Mandatory = false,
+            HelpMessage = "The name of the virtual router.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string RouterName { get; set; }
+
+        [Parameter(
+            ParameterSetName = VirtualRouterParameterSetNames.ByVirtualRouterResourceId,
+            Mandatory = true,
+            HelpMessage = "ResourceId of the virtual router.",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (string.Equals(this.ParameterSetName, VirtualRouterParameterSetNames.ByVirtualRouterResourceId, StringComparison.OrdinalIgnoreCase))
+            {
+                var resourceInfo = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceInfo.ResourceGroupName;
+                RouterName = resourceInfo.ResourceName;
+            }
+
+            if (ShouldGetByName(ResourceGroupName, RouterName))
+            {
+                var vVirtualRouter = this.NetworkClient.NetworkManagementClient.VirtualRouters.Get(ResourceGroupName, RouterName);
+                var vVirtualRouterModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSVirtualRouter>(vVirtualRouter);
+                vVirtualRouterModel.ResourceGroupName = this.ResourceGroupName;
+                vVirtualRouterModel.Tag = TagsConversionHelper.CreateTagHashtable(vVirtualRouter.Tags);
+
+                if (vVirtualRouter.Peerings != null && vVirtualRouter.Peerings.Count > 0)
+                {
+                    var vVirtualRouterPeering = this.NetworkClient.NetworkManagementClient.VirtualRouterPeerings.List(ResourceGroupName, RouterName);
+                    var vVirtualRouterPeeringList = ListNextLink<VirtualRouterPeering>.GetAllResourcesByPollingNextLink(vVirtualRouterPeering,
+this.NetworkClient.NetworkManagementClient.VirtualRouterPeerings.ListNext);
+                    foreach (var peering in vVirtualRouterPeeringList)
+                    {
+                        vVirtualRouterModel.Peerings.Add(NetworkResourceManagerProfile.Mapper.Map<CNM.PSVirtualRouterPeer>(peering));
+                    }
+                }
```

Comment:
I think you can make these code into a function then you can reuse them in both conditions.

Reply:
Done. Defined a method in VirtualRouterBaseCmdlet